src/unix/process.c

```diff
@@ -383,6 +391,74 @@ static void uv__process_child_init(const uv_process_options_t* options,
 #endif
 
 
+int uv_get_children_pid(pid_t ppid, uint32_t** proc_list, int* proc_count) {
+  uint32_t* temp = uv__malloc(0);
+  size_t len = 0;
+
+#if defined(__APPLE__)    || \
+    defined(__NetBSD__)   || \
+    defined(__OpenBSD__)  || \
+    defined(__FreeBSD__)  || \
+    defined(__DragonFly__)
+  struct kinfo_proc *p_list = NULL;
+  int ret, p_count, i;
+  /* ref:
+     http://unix.superglobalmegacorp.com/Net2/newsrc/sys/kinfo_proc.h.html */
+  static const int name[] = { CTL_KERN, KERN_PROC, KERN_PROC_ALL, 0 };
+
+  *proc_list = NULL;
+  *proc_count = 0;
+  /* get number of total processes for subsequent calls */
+  ret = sysctl((int *)name, (sizeof(name) / sizeof(*name)) - 1, NULL, &len, NULL, 0);
+  if (ret) return 1;
+
+  p_list = malloc(len);
+  if (!p_list) return 1;
+  /* get the process list */
+  ret = sysctl((int *)name, (sizeof(name) / sizeof(*name)) - 1, p_list, &len, NULL, 0);
+  if (ret) return 1;
+
+  p_count = len / sizeof(struct kinfo_proc);
+  /* iterate though whole p_list */
+  for (i = 0; i < p_count; i++) {
+   /* if parent is ppid, push pid to array; increase counter */
+   if ((uint32_t) p_list[i].kp_eproc.e_ppid == (uint32_t) ppid) {
+     uv__realloc(temp, (*proc_count + 1) * sizeof(uint32_t));
+     temp[*proc_count] = (uint32_t)p_list[i].kp_proc.p_pid;
+
+     (*proc_count)++;
+   }
+  }
+  free(p_list);
+
+#elif defined(__linux__)
+  char *line = NULL;
+  char proc_p[256] = {0};
+  int *fp;
+
+  *proc_list = NULL;
+  *proc_count = 0;
+  /* Rationale: children are defined in thread with sames ID of process -> read,
+   * check line endings, count, build array, return */
+  sprintf(proc_p, "/proc/%u/task/%u/children", ppid, ppid);
+  fp = fopen(proc_p,"r");
+  if (fp == NULL)
+    return 127;
+
+  while (getline(&line, &len, fp) >= 0) {
```

Comment:
At least for me (Linux 4.2.0-35-generic x86_64), the child pids are not newline-separated, but separated by spaces, so I think `fscanf` is better here (it might be anyway because it let’s you specify the format so you don’t have to allocate stuff)?
If not, keep in mind `getline` allocates the memory pointed to by `line` if it’s `NULL`, so you’d have to free it and reset it and `len` to zero.